checker/manual/creating-a-checker.tex

```diff
@@ -1094,34 +1092,136 @@ \section{The checker class:  Compiler interface\label{writing-compiler-interface
 \refmethodterse{framework/type}{AnnotatedTypeFactory}{createSupportedTypeQualifiers}{--}
-method.  Each argument to \<@TypeQualifiers> or value returned by
-\<createSupportedTypeQualifiers> is a class literal for a type qualifier
-whose definition bears the
-\refqualclass{framework/qual}{TypeQualifier} meta-annotation.
+method in one of the ways shown below.
+
+Note that \code{createSupportedTypeQualifiers()} returns an immutable set of the
+annotation classes supported by a checker.
+
+In total, there are 6 ways to indicate annotations that are supported by a
+checker:
+\begin{enumerate}
+\item Only support annotations located in a checker's \<qual> directory, and
+\<PolyAll>:
+
+This is the default behavior. Simply place those annotations within the \<qual>
+directory.
+
+\item Support annotations located in a checker's \<qual> directory, but without
+\<PolyAll>:
+
+Place those annotations within the \<qual>
+directory, and override \<createSupportedTypeQualifiers()> by calling
+\<getBundledTypeQualifiersWithoutPolyAll()> with no parameters passed in.
+Code example:
+
+\begin{Verbatim}
+@Override
+protected Set<Class<? extends Annotation>> createSupportedTypeQualifiers() {
+    return getBundledTypeQualifiersWithoutPolyAll();
+}
+\end{Verbatim}
+
+\item Support annotations located in a checker's \<qual> directory, \<PolyAll>,
+and a list of other annotations:
+
+Place those annotations within the \<qual>
+directory, and override \<createSupportedTypeQualifiers()> by calling
+\<getBundledTypeQualifiersWithPolyAll()> with a varargs parameter list of the
+other annotations. Code example:
+
+\begin{Verbatim}
+@Override
+protected Set<Class<? extends Annotation>> createSupportedTypeQualifiers() {
+    return getBundledTypeQualifiersWithPolyAll(
+            Regex.class, PartialRegex.class,
+            RegexBottom.class, UnknownRegex.class);
+}
+\end{Verbatim}
+
+\item Support annotations located in a checker's \<qual> directory and a list
+of other annotations, but without supporting \<PolyAll>:
+
+Place those annotations within the \<qual>
+directory, and override \<createSupportedTypeQualifiers()> by calling
+\<getBundledTypeQualifiersWithoutPolyAll()> with a varargs parameter list of the
+other annotations. Code example:
+
+\begin{Verbatim}
+@Override
+protected Set<Class<? extends Annotation>> createSupportedTypeQualifiers() {
+    return getBundledTypeQualifiersWithoutPolyAll(
+            UnknownFormat.class, FormatBottom.class);
+}
+\end{Verbatim}
+
+\item Supporting only annotations that are explicitly listed, and \<PolyAll>:
+
+Override \<createSupportedTypeQualifiers()> and return an immutable set of the
+supported annotations, plus \<PolyAll>. Code example:
+\begin{Verbatim}
+@Override
+protected Set<Class<? extends Annotation>> createSupportedTypeQualifiers() {
+return Collections.unmodifiableSet(
+    new HashSet<Class<? extends Annotation>>(
+        Arrays.asList(
+            A.class, B.class,
+            PolyAll.class)));
```

Comment:
Is this really worth separating into two possibilities? You have to list PolyAll explicitly, so there is no different method that you call to automatically support it.